dali/python/nvidia/dali/external_source.py

```diff
@@ -222,9 +222,14 @@ def _is_generator_function(x):
     where __call__ is a generator function"""
     if inspect.isgeneratorfunction(x):
         return True
+    if isinstance(x, functools.partial):
+        return _is_generator_function(x.func)
```

Comment:
The field/property `func` comes from:
https://docs.python.org/3/library/functools.html#functools.partial